Makefile

```diff
@@ -15,13 +15,19 @@ NXDK_STACKSIZE = 65536
 endif
 
 UNAME_S := $(shell uname -s)
+UNAME_M := $(shell uname -m)
+
 ifeq ($(UNAME_S),Linux)
 LD           = lld -flavor link
 LIB          = llvm-lib
 AS           = clang
 CC           = clang
 CXX          = clang++
+ifneq ($(UNAME_M),x86_64)
+CGC          = $(NXDK_DIR)/tools/cg/linux/cgc.i386
```

Comment:
I'd suggest to instead move the files around, so we have `linux/x86_64/cgc` and `linux/???/cgc` for the i386 variant. We can then probably just use `$(NXDK_DIR)/tools/cg/linux/$(UNAME_M)/cgc`.
This will also lead to meaningful error messages.

Extending the original filename sounds non-standard and confusing.

I don't feel too strongly but if this is not done in this PR, then I'd probably open an issue about this.

Reply:
Let's do it later, as the other platforms don't have such directory structure yet

Reply:
I think this is primarily a Linux issue, so I'd be fine if this is on Linux. I believe macOS is exclusively x64 now, and I *assume* the existing Windows binary is x86 which also works on x64?